apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/PartitionOptions.cs

```diff
@@ -0,0 +1,84 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using System.Threading;
+
+namespace Google.Cloud.Spanner.Data;
+
+/// <summary>
+/// An immutable class representing partition options.
+/// The specified values of `PartitionSizeBytes`, `MaxPartitions` and `CancellationToken`
+/// are utilized in the RPC call while generating the partition tokens.
+/// The specified flag for `DataBoostEnabled` will be stored in the request property
+/// of the <see cref="CommandPartition"/> object and is used when executing the
+/// partition object. If `DataBoostEnabled` is set to true, the request will be
+/// executed using Spanner independent compute resources.
+/// </summary>
+public sealed class PartitionOptions
+{
+    /// <summary>
+    /// Returns a default instance of <see cref="PartitionOptions"/> with databoost enabled flag set to false.
+    /// </summary>
+    public static PartitionOptions Default { get; } = new PartitionOptions(null, null, false);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the databoost enabled flag updated to the provided value.
+    /// </summary>
+    /// <returns> A new instance of <see cref="PartitionOptions"/> class.</returns>
+    public PartitionOptions WithDataBoostEnabled(bool? dataBoostEnabled) => new(PartitionSizeBytes, MaxPartitions, dataBoostEnabled);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the partition size bytes updated to the provided value.
+    /// </summary>
+    /// <returns> A new instance of <see cref="PartitionOptions"/> class.</returns>
+    public PartitionOptions WithPartitionSizeBytes(long? partitionSizeBytes) => new(partitionSizeBytes, MaxPartitions, DataBoostEnabled);
+
+    /// <summary>
+    /// Returns a new instance with the same options as this one, but with the max partitions updated to the provided value.
+    /// </summary>
+    /// <returns> A new instance of <see cref="PartitionOptions"/> class.</returns>
+    public PartitionOptions WithMaxPartitions(long? maxPartitions) => new(PartitionSizeBytes, maxPartitions, DataBoostEnabled);
+
+    /// <summary>
+    /// The desired data size for each partition generated.  The default for this
+    /// option is currently 1 GiB.  This is only a hint. The actual size of each
+    /// partition may be smaller or larger than this size request.
+    /// </summary>
+    public long? PartitionSizeBytes { get; }
+
+    /// <summary>
+    /// The desired maximum number of partitions to return.  For example, this may
+    /// be set to the number of workers available.  The default for this option
+    /// is currently 10,000. The maximum value is currently 200,000.  This is only
```

Comment:
The default for this property is null, and I think we should say so - and say that that means the server gets to decide. We *might* want to document the currently defaulting, but I'm not sure whether it's worth it (and it could easily get out of date).

Ditto for PartitionSizeBytes.